### PR TITLE
Identify critical use cases

### PR DESCRIPTION
**Pull request by jjanvier**
### What's in this PR?

- Some Behats and PHPUnits have been tagged as _"critical"_. **They identify what's absolutely vital for the PIM to work.**
- A few dupplicated Behats have been removed 

### How has it been done?

With the cooperation of our dear product managers. They told us what was critical, and we found the related tests. 

### A few weirdnesses

Two important things to keep in mind:

1. **This PR does not represent the truth. This is only the knowledge we had at a given time.** So, if you think some critical tests are missing, or some are useless, that's perfect. Talk with your team and your product owner to identify what's essential or not for the PIM to work.

2. Sometimes, we tagged "bad" Behats. By bad I mean:

- the test has many steps and only a sub-part of them is critical (but we didn't find the check of this behavior elsewhere)
- the test actually checks the behavior we want, but in a way that could/should be reworked

### OK, what for?

This work offers us two important things:

1. _Maybe_ we'll be able to shorten our builds by launching only the test suites we work on + the critical tests. For instance, if I work on the Assets, the assets suite will be launched, as well as all the critical tests. This should mitigate the risks about merging broken stuff. I'm working on a PR that tries to launch only critical tests, to see where it goes and if it's useful from a build time point of view.

2. It will also help us to know what we should focus on when rewriting Behats. Those tests will be the first candidates to learn about how to write correctly use cases with Gherkin.

### A few figures

|  | critical tests | all (no specs) | percentage |
| --- | --- | --- | --- |
| *CE phpunits* | 45 | 2457 | 1.8% | 
| *EE phpunits* | 32 | 779 | 4.1% |
| *CE behats* | 120 | 1353 | 8.9% |
| *EE behats* | 51 | 876 | 5.8% |
| **overall** | 248 | 5465 | 4.5% |

### CODE REVIEW

**Comment by jjanvier** [comment]
TODO:
- where should we put the `product groups`? in enrichment or structure. See with Delph.

File: tests/back/Pim/Enrichment/EndToEnd/Category/ExternalApi/CreateCategoryEndToEnd.php [hunk 1]
```diff
@@ -53,6 +53,9 @@ public function testStandardFormatWhenACategoryIsCreatedButIncompleted()
         $this->assertSame($categoryStandard, $normalizer->normalize($category));
     }
 
+    /**
+     * @critical
```
Comment (jjanvier) [inline tests/back/Pim/Enrichment/EndToEnd/Category/ExternalApi/CreateCategoryEndToEnd.php]: add `@group`

File: tests/back/Pim/Enrichment/EndToEnd/Category/ExternalApi/PartialUpdateCategoryEndToEnd.php [hunk 2]
```diff
@@ -95,6 +95,9 @@ public function testStandardFormatWhenACategoryIsCreatedWithAnEmptyContent()
         $this->assertSame($categoryStandard, $normalizer->normalize($category));
     }
 
+    /**
+     * @criticial
```
Comment (jjanvier) [inline tests/back/Pim/Enrichment/EndToEnd/Category/ExternalApi/PartialUpdateCategoryEndToEnd.php]: add `@group critical`

File: tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/Media/ListMediaFileEndToEnd.php [hunk 3]
```diff
@@ -16,6 +16,9 @@ public function setUp(): void
         $this->createMedia(new \SplFileInfo($this->getFixturePath('akeneo.txt')));
     }
 
+    /**
+     * @critical
```
Comment (jjanvier) [inline tests/back/Pim/Enrichment/EndToEnd/Product/Product/ExternalApi/Media/ListMediaFileEndToEnd.php]: add `@group`

File: tests/legacy/features/pim/enrichment/product-model/import/update_product_models_when_importing_families.feature [hunk 4]
```diff
@@ -3,6 +3,10 @@ Feature: Update product models when importing families
   As a product manager
   I need to update the product models when families are imported
 
+  # should be more precise:
+  # - attribute as required for the completeness (or not)
+  # - adding an attribute to the family
+  @critical
```
Comment (jjanvier) [inline tests/legacy/features/pim/enrichment/product-model/import/update_product_models_when_importing_families.feature]: one line

File: tests/legacy/features/pim/structure/family/define_the_attribute_requirement.feature [hunk 5]
```diff
@@ -34,6 +36,7 @@ Feature: Define the attribute requirement
     And I visit the "Attributes" tab
     Then attribute "description" should not be required in channels mobile and tablet
 
+    @TODO move to enrichment
```
Comment (jjanvier) [inline tests/legacy/features/pim/structure/family/define_the_attribute_requirement.feature]: # todo

File: tests/legacy/features/pim/enrichment/category/create_a_category.feature [hunk 6]
```diff
@@ -2,12 +2,13 @@
 Feature: Create a category
   In order to provide a tree of my product categories
   As a product manager
-  I need to be able to create a new category tree or a node of a category tree
+  I need to be able to create a new category tree or a sub-category
```
Comment (arnolanglade) [inline tests/legacy/features/pim/enrichment/category/create_a_category.feature]: is it normal ?

Comment (jjanvier) [reply tests/legacy/features/pim/enrichment/category/create_a_category.feature]: ah yes, the wording was bad

Comment (arnolanglade) [reply tests/legacy/features/pim/enrichment/category/create_a_category.feature]: yes you're right: I thought it was not a step... :palm_face:

**Pull request merged**